src/java.base/share/classes/java/nio/channels/AsynchronousFileChannel.java

```diff
@@ -425,10 +425,13 @@ public static AsynchronousFileChannel open(Path file, OpenOption... options)
      * required then a region starting at zero, and no smaller than the
      * expected maximum size of the file, should be locked.  The two-argument
      * {@link #lock(Object,CompletionHandler)} method simply locks a region
-     * of size {@link Long#MAX_VALUE}. If a lock that overlaps the requested
-     * region is already held by this Java virtual machine, or this method has
-     * been invoked to lock an overlapping region and that operation has not
-     * completed, then this method throws {@link OverlappingFileLockException}.
+     * of size {@link Long#MAX_VALUE}. If a lock is created with a valid
+     * {@code position} and a {@code size} parameter of zero, then a lock of
```

Comment:
I think we've got the javadoc change to a good place. One minor point is that "created with a valid position" may be a bit confusing as you can't create a lock with an invalid position. What would you think of "If the position is non-negative and the size is zero then a lock of size ...".

Reply:
That would likely be better; will change.